agent-control/src/version_checker/handler.rs

```diff
@@ -0,0 +1,43 @@
+use crate::sub_agent::error::SubAgentError;
+use crate::version_checker::AgentVersion;
+use opamp_client::StartedClient;
+use opamp_client::opamp::proto::{AnyValue, KeyValue, any_value};
+
+/// This method request the AgentDescription from the current opamp client and, updates or add the
+/// field from agent version to be sent to opamp server
+pub fn set_agent_description_version<C>(
+    version: AgentVersion,
+    maybe_opamp_client: Option<&C>,
+) -> Result<(), SubAgentError>
+where
+    C: StartedClient,
+{
+    if let Some(client) = maybe_opamp_client.as_ref() {
```

Comment:
nit: we can exit early if there is no opamp and reduce the indentation

Reply:
I don't quite like that a function is a no-op just because of its arguments happens to be at some value (`None` is a value!) so I reworked it to be a function that I can `map` over the `Option` instead. WDYT?

Reply:
One time I'll bring the issues with having logic inside select! macros (formatting is awful for starters) to a tech talk again 🤣

Just to extract these to private functions if we can!